File: javascript/test/Structural/Adapter.js
(function() {
  define(['Structural/Adapter/TeaBag', 'Structural/Adapter/TeaCup', 'Structural/Adapter/TeaBall', 'Structural/Adapter/LooseLeafTea'], function(TeaBag, TeaCup, TeaBall, LooseLeafTea) {
    'use strict';
    return describe('Test Structural Adapter', function() {
      it('Steeping tea bag', function() {
        var teaBag, teaCup;
        teaCup = new TeaCup();
        teaBag = new TeaBag();
        return expect(teaCup.steepTeaBag(teaBag)).toEqual("tea bag is steeping in cup");
      });
      return it('Steeping loose leaf tea', function() {
        var looseLeafTea, teaBall, teaCup;
        teaCup = new TeaCup();
        looseLeafTea = new LooseLeafTea();
        teaBall = new TeaBall(looseLeafTea);
        return expect(teaCup.steepTeaBag(teaBall)).toBeTruthy();
      });
    });
  });

}).call(this);

File: scala/src/test/scala/Structural/Adapter.scala
package test.Structural;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Structural.Adapter.*;



//TestTeaBagAdaptation - testing the adapter

public class Adapter {

	@Test
   public void thetester() {
		

       TeaCup teaCup = new TeaCup();       
       TeaBag teaBag = new TeaBag();       
       teaCup.steepTeaBag(teaBag);
       org.junit.Assert.assertTrue("Steeping tea bag",
			teaBag.teaBagIsSteeped );

       LooseLeafTea looseLeafTea = new LooseLeafTea();
       TeaBall teaBall = new TeaBall(looseLeafTea);
       teaCup.steepTeaBag(teaBall);
       org.junit.Assert.assertTrue("Steeping loose leaf tea",
				looseLeafTea.teaIsSteeped
		);
   }
}      
